src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -0,0 +1,133 @@
+use Stripe\Event as StripeEvent;
+
+class StripeUpdate extends Action
+{
+    private DonationRepository $donationRepository;
+    private EntityManagerInterface $entityManager;
+    private SerializerInterface $serializer;
+
+    public function __construct(
+        DonationRepository $donationRepository,
+        EntityManagerInterface $entityManager,
+        LoggerInterface $logger,
+        SerializerInterface $serializer
+    ) {
+        $this->donationRepository = $donationRepository;
+        $this->entityManager = $entityManager;
+        $this->serializer = $serializer;
+
+        parent::__construct($logger);
+    }
+
+    protected function action(): Response
+    {
+        $payload = $this->request->getBody();
+        $signature = $this->request->getHeaderLine('stripe-signature');
+        $webhookSecret = 'whsec_V4mcupoQsVuKmP8PuTMIAHf83E8LBRHt'; // TODO replace with one supplied in Infra
+
+        try {
+            $event = \Stripe\Webhook::constructEvent($payload, $signature, $webhookSecret);
+        } catch (\UnexpectedValueException $e) {
+            $logMessage = 'Invalid Payload';
+            $this->logger->warning($logMessage);
+            $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+            return $this->respond(new ActionPayload(400, null, $error));
+        } catch (\Stripe\Exception\SignatureVerificationException $e) {
+            $logMessage = 'Invalid Signature';
+            $this->logger->warning($logMessage);
+            $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+            return $this->respond(new ActionPayload(400, null, $error));
+        }
+
+        /** @var Donation $donation */
+        $donation = $this->donationRepository->findOneBy(['transactionId' => $event->data->object->id]);
+
+        if ($event instanceof StripeEvent && $donation) {
+            // Handle the event
+            switch ($event->type) {
+                case 'payment_intent.succeeded':
+                    $this->handlePaymentIntentSucceeded($event, $donation);
+                    break;
+                case 'payment_intent.created':
+                    $logMessage = 'Unsupported Action';
+                    $this->logger->warning($logMessage);
+                    $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+                    return $this->respond(new ActionPayload(400, null, $error));
+                default:
+                    $logMessage = 'Unsupported Action';
+                    $this->logger->warning($logMessage);
+                    $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+                    return $this->respond(new ActionPayload(400, null, $error));
+            }
+        } else {
+            $logMessage = 'No Content';
+            $this->logger->warning($logMessage);
+            $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
```

Comment:
Not a bad request. The body should be an empty string in line with the 'No Content' code if we're going to follow this convention.